crates/stoffel-vm/src/mpc_builtins.rs

```diff
@@ -0,0 +1,1510 @@
+    pub fn is_share_object(store: &ObjectStore, value: &Value) -> bool {
+        match value {
+            Value::Share(_, _) => true,
+            Value::Object(_) => store
+                .get_field(value, &Value::String(share_fields::TYPE.to_string()))
+                .map(|v| v == Value::String(share_fields::TYPE_VALUE.to_string()))
+                .unwrap_or(false),
+            _ => false,
+        }
+    }
+
+    /// Get the ShareType from a Share object
+    pub fn get_share_type(store: &ObjectStore, value: &Value) -> Result<ShareType, String> {
+        match value {
+            Value::Share(ty, _) => Ok(*ty),
+            Value::Object(_) => {
+                let share_type_field = store
+                    .get_field(value, &Value::String(share_fields::SHARE_TYPE.to_string()))
+                    .ok_or_else(|| "Share object missing __share_type field".to_string())?;
+
+                match share_type_field {
+                    Value::String(s) if s == share_fields::SECRET_INT => {
+                        let bit_length = store
+                            .get_field(value, &Value::String(share_fields::BIT_LENGTH.to_string()))
+                            .and_then(|v| match v {
+                                Value::I64(n) => Some(n as usize),
+                                _ => None,
+                            })
+                            .unwrap_or(64);
+                        Ok(ShareType::SecretInt { bit_length })
+                    }
+                    Value::String(s) if s == share_fields::SECRET_FIXED_POINT => {
+                        let k = store
+                            .get_field(value, &Value::String(share_fields::PRECISION_K.to_string()))
+                            .and_then(|v| match v {
+                                Value::I64(n) => Some(n as usize),
+                                _ => None,
+                            })
+                            .unwrap_or(64);
+                        let f = store
+                            .get_field(value, &Value::String(share_fields::PRECISION_F.to_string()))
+                            .and_then(|v| match v {
+                                Value::I64(n) => Some(n as usize),
+                                _ => None,
+                            })
+                            .unwrap_or(16);
+                        Ok(ShareType::secret_fixed_point_from_bits(k, f))
+                    }
+                    _ => Err(format!("Unknown share type: {:?}", share_type_field)),
+                }
+            }
+            _ => Err(format!("Not a Share object: {:?}", value)),
+        }
+    }
+}
+
+/// Register all MPC builtin functions with the VM
+pub fn register_mpc_builtins(vm: &mut VirtualMachine) {
+    register_share_builtins(vm);
+    register_mpc_info_builtins(vm);
+    register_rbc_builtins(vm);
+    register_aba_builtins(vm);
+    register_consensus_builtins(vm);
+    register_federated_learning_builtins(vm);
```

Comment:
We don't need an FL built-in because this will be implemented via the language. Further, you haven't defined the fields for it in this file either.